syntax = "proto3";

package keyvalue;

service KVStore {

  rpc GetKey (KeyRequest) returns (KeyValueReply) {}

  rpc PutKey (KeyValueRequest) returns (KeyValueReply) {}

  rpc DeleteKey (KeyRequest) returns (KeyValueReply) {}
}

message KeyRequest {
  string key = 1;
  int64 timestamp = 2;
}

message KeyValueRequest {
  string key = 1;
  string value = 2;
  int64 timestamp = 3;
}

message KeyValueReply {
  string key = 1;
  string value = 2;
  string message = 3;
  int32 status = 4;
}
